quesma/schema/registry.go

```diff
@@ -74,6 +81,56 @@ func (s *schemaRegistry) getInternalToPublicFieldEncodings(tableName string) map
 	return internalToPublicFieldsEncodings
 }
 
+func (s *schemaRegistry) invalidateCache() {
+	s.cachedSchemas = nil
+}
+
+func (s *schemaRegistry) start() {
+
+	notificationChannel := make(chan types.ReloadMessage, 1)
+
+	s.dataSourceTableProvider.RegisterTablesReloadListener(notificationChannel)
+
+	protectedReload := func() {
+		defer recovery.LogPanic()
+		s.Lock()
+		defer s.Unlock()
+
+		s.invalidateCache()
+	}
+
+	go func() {
```

Comment:
This starts a new goroutine. I think we should also provide a stop method for graceful shutdown.